subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7DirectoryWalkerTest.groovy

```diff
@@ -319,4 +324,36 @@ class Jdk7DirectoryWalkerTest extends Specification {
         }
     }
 
+    def "directory snapshotter returns the same details as directory walker"() {
+        given:
+        def rootDir = tmpDir.createDir("root")
+        generateFilesAndSubDirectories(rootDir, 10, 5, 3, 1, new AtomicInteger(0))
+        MirrorUpdatingDirectoryWalker directorySnapshotter = new MirrorUpdatingDirectoryWalker(TestFiles.fileHasher(), TestFiles.fileSystem(), new StringInterner())
+        def patternSet = Mock(PatternSet)
+        List<FileVisitDetails> visitedWithJdk7Walker = null
```

Comment:
No need to set this to `null`. You could actually do the walking here, as it is a prerequisite to the `when:` part, not part of it.